src/datasources/positions-api/zerion-positions-api.service.ts

```diff
@@ -0,0 +1,242 @@
+  private readonly limitCalls: number;
+
+  constructor(
+    @Inject(CacheService) private readonly cacheService: ICacheService,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(NetworkService) private readonly networkService: INetworkService,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    private readonly httpErrorFactory: HttpErrorFactory,
+  ) {
+    this.apiKey = this.configurationService.get<string>(
+      'balances.providers.zerion.apiKey',
+    );
+    this.baseUri = this.configurationService.getOrThrow<string>(
+      'balances.providers.zerion.baseUri',
+    );
+    this.defaultExpirationTimeInSeconds =
+      this.configurationService.getOrThrow<number>(
+        'expirationTimeInSeconds.default',
+      );
+    this.defaultNotFoundExpirationTimeSeconds =
+      this.configurationService.getOrThrow<number>(
+        'expirationTimeInSeconds.notFound.default',
+      );
+    this.chainsConfiguration = this.configurationService.getOrThrow<
+      Record<number, ChainAttributes>
+    >('balances.providers.zerion.chains');
+    this.fiatCodes = this.configurationService
+      .getOrThrow<Array<string>>('balances.providers.zerion.currencies')
+      .map((currency) => currency.toUpperCase());
+    this.limitPeriodSeconds = configurationService.getOrThrow(
+      'balances.providers.zerion.limitPeriodSeconds',
+    );
+    this.limitCalls = configurationService.getOrThrow(
+      'balances.providers.zerion.limitCalls',
+    );
+  }
+
+  async getPositions(args: {
+    chain: Chain;
+    safeAddress: `0x${string}`;
+    fiatCode: string;
+  }): Promise<Raw<Array<Position>>> {
+    if (!this.fiatCodes.includes(args.fiatCode.toUpperCase())) {
+      throw new DataSourceError(
+        `Unsupported currency code: ${args.fiatCode}`,
+        400,
+      );
+    }
+
+    const cacheDir = CacheRouter.getZerionPositionsCacheDir({
+      chainId: args.chain.chainId,
+      safeAddress: args.safeAddress,
+      fiatCode: args.fiatCode,
+    });
+    const chainName = this._getChainName(args.chain);
+    const cached = await this.cacheService.hGet(cacheDir);
+    if (cached != null) {
+      const { key, field } = cacheDir;
+      this.loggingService.debug({ type: LogType.CacheHit, key, field });
+      const zerionBalances = z
+        .array(ZerionBalanceSchema)
+        .parse(JSON.parse(cached));
+      return this._mapPositions(chainName, zerionBalances);
+    }
+
+    try {
+      await this._checkRateLimit();
+      const { key, field } = cacheDir;
+      this.loggingService.debug({ type: LogType.CacheMiss, key, field });
+      const url = `${this.baseUri}/v1/wallets/${args.safeAddress}/positions`;
+      const networkRequest = {
+        headers: { Authorization: `Basic ${this.apiKey}` },
+        params: {
+          'filter[chain_ids]': chainName,
+          'filter[positions]': 'only_complex',
```

Comment:
This file is a 1-1 copy of `zerion-balances-api.service.ts` with the exception of this line as we only care about defi positions and not the wallet balances. I also removed some functions that were not needed.